File: content/snippets/python/language_basics/json_and_serialization.py
#!/usr/bin/env python3
import json


def main() -> None:
    user = {"name": "Alice", "age": 29}
    text = json.dumps(user)
    parsed = json.loads(text)

    print("json:", text)
    print("parsed name:", parsed["name"])


if __name__ == "__main__":
    main()

File: content/snippets/go/language_basics/json_and_serialization.go
package main

import (
	"encoding/json"
	"fmt"
)

type User struct {
	Name string `json:"name"`
	Age  int    `json:"age"`
}

func main() {
	user := User{Name: "Alice", Age: 29}
	data, err := json.Marshal(user)
	if err != nil {
		panic(err)
	}

	var parsed User
	if err := json.Unmarshal(data, &parsed); err != nil {
		panic(err)
	}

	fmt.Println("json:", string(data))
	fmt.Println("parsed name:", parsed.Name)
}
